reccmp/tools/asmcmp.py

```diff
@@ -221,9 +223,17 @@ def main():
 
     logging.basicConfig(level=args.loglevel, format="[%(levelname)s] %(message)s")
 
-    with Bin(target.original_path, find_str=True) as origfile, Bin(
-        target.recompiled_path
-    ) as recompfile:
+    origfile = detect_image(filepath=target.original_path)
```

Comment:
Since there's only one argument to `detect_image`, we can probably drop the filepath key here and in other scripts. (Or use it consistently if the intent is to future-proof)

Reply:
I think it'd be better to change the API by splitting binary format detection and loading.
Right now, `detect_file` does more then detecting: it also loads the file.